cGame_logic.cpp

```diff
@@ -645,9 +651,10 @@ void cGame::shutdown() {
     }
 
     // Destroy font of Allegro FONT library
+#ifdef D2TM_DEFAULTS
```

Comment:
I know why you do this (as you said it breaks on linux). But I do wonder why it breaks there. Is it because of alfont itself? Or perhaps the resources it tries to destroy (`game_font` and `bene_font`) are somehow no longer valid here?

Perhaps if you can find out the reason why this is needed, we can add a comment for that. 

In the end, when migrating to SDL2 this will be removed anyway :)

Reply:
I investigate a little more. `alfont_destroy_font` crash without explication. Normally, I would have been more conscientious, memory leaks are bad. But since alfont is old, deprecated, not well documented, and we're going to replace it, I won't spend any more time on it.

Reply:
Perhaps we don't need to destrouy the font at all?
We don't destroy small_font, but only these two.
The next line is, alfont_exit. I hope that would destroy fonts. I couldn't find any documentation...

I would suggest commenting out the delete_font lines (and adding a commented out one about small_font as well) with a comment above explaining that this crashes and that we think it is not needed anyway.

Then we don't need that OS define. Ideally, we would have no OS specific code in the d2tm code at all.

Reply:
good point, if we don't destroy the font at all we can leave out the `#ifdef` part. And then we have best of both worlds.